sdk/iot/provisioning/samples/src/paho_iot_provisioning_example.c

```diff
@@ -262,6 +220,119 @@ static int register_device()
   return 0;
 }
 
+static void print_buffer(az_span span)
+{
+  char* buffer = (char*)az_span_ptr(span);
+  for (int i = az_span_size(span); i > 0; i--)
+  {
+    putchar(*buffer++);
+  }
+
+  putchar('\n');
+}
+
+static int get_operation_status()
+{
+  int rc;
+  bool registration_complete = false;
+  char* topicName;
+  int topicLen;
+  MQTTClient_message* message;
+
+  while (!registration_complete)
+  {
+    if ((rc = MQTTClient_receive(mqtt_client, &topicName, &topicLen, &message, 60 * 1000))
+        != MQTTCLIENT_SUCCESS)
+    {
+      printf("Failed to receive message, return code %d\n", rc);
+      return rc;
+    }
+
+    if (topicLen == 0)
+    {
+      // The length of the topic if there are one more NULL characters embedded in topicName,
+      // otherwise topicLen is 0.
+      topicLen = (int)strlen(topicName);
+    }
+
+    printf("Message arrived\n");
+    printf("     topic (%d): %s\n", topicLen, topicName);
+    printf("   message (%d): ", message->payloadlen);
+    print_buffer(az_span_init(message->payload, message->payloadlen));
+
+    az_iot_provisioning_client_register_response response;
+    char query_topic[256];
```

Comment:
Can we move all these buffer allocated on stack to data segment?

Reply:
I don't think that would be a good idea for scalability: stack allocation means that the memory isn't always in-use. The reservation is made only within this stack frame.
Having these global means: 
1. We cannot have 2 registrations in parallel.
2. The memory is always allocated even if not used.